dvc/repo/run.py

```diff
@@ -91,32 +38,13 @@ def run(self, fname=None, no_exec=False, single_stage=False, **kwargs):
     if not stage_name and not single_stage:
         raise InvalidArgumentError("`-n|--name` is required")
 
-    if single_stage:
-        kwargs.pop("name", None)
-        stage_cls = Stage
-        path = fname or _get_file_path(kwargs)
-    else:
-        if not is_valid_name(stage_name):
-            raise InvalidStageName
-
-    params = chunk_dict(parse_params_from_cli(kwargs.pop("params", [])))
-    stage = create_stage(
-        stage_cls, repo=self, path=path, params=params, **kwargs
+    stage = create_stage_from_cli(
+        self, single_stage=single_stage, fname=fname, **kwargs
     )
-    restore_meta(stage)
     if kwargs.get("run_cache", True) and stage.can_be_skipped:
         return None
 
-    dvcfile = Dvcfile(self, stage.path)
-    try:
-        if kwargs.get("force", True):
-            with suppress(ValueError):
-                self.stages.remove(stage)
-        else:
-            _check_stage_exists(dvcfile, stage)
-        self.check_modified_graph([stage])
-    except OutputDuplicationError as exc:
-        raise OutputDuplicationError(exc.output, set(exc.stages) - {stage})
+    check_graphs(self, stage, force=kwargs.get("force", True))
```

Comment:
I don't feel good about this, but the `can_be_skipped` checks make it difficult to keep it within `create_stage_from_cli`.

Reply:
@skshetry Yeah, was thinking about dropping that logic in 2.0 too.